.drone.yml

```diff
@@ -1607,7 +1815,6 @@ platform:
 
 steps:
   - name: manifest-rootless
-    pull: always
```

Comment:
This was set twice in the same step, see two lines below.

Reply:
`pull: always` should only be set for the first occurence of an image in the whole file. @jolheiser can you check that this is still true?

Reply:
I can double-check, sure.  
Since some of these can now run in parallel, it might be a bit off. I'll see what comes up.

Reply:
Yeah I guess for the sake of performance, we can skip double pull in parallel runs. The CI runs often enough to keep images up to date. Even better would be an outside job on the server that updates images once a day or so.

Reply:
A cron (pipeline) would be an interesting solution to this, and would probably offset the image pulling load a fair bit.  
I think that should likely be in a separate PR, but for this one I will at least make sure that it's not doubling up on `always` pulls.

Reply:
@silverwind many of these were indeed copy/pasted as I split them up, but even other than that there's a _ton_ of duplicates that are all pulling always. 

Is it worth cleaning them all up in this PR, or would a follow-up make sense?

Reply:
Assuming this is indeed required for the entire file vs each pipeline.

Each pipeline can use a separate agent/runner, so should it actually just be unique per-pipeline instead of the entire file?

Reply:
Depends on whether the pipelines run on same docker daemon or not. I guess it could be distributed across multiple machines, in which case you are correct, pulls need to be per-pipeline, so it's better to keep it like that.

Reply:
Currently we do use multiple machines (@techknowlogick knows more specifics), so I think a follow-up might be better served for cleaning up image pulling.

Reply:
Generally I think this whole image pulling business is something that drone could do itself. Even if you make a cron pipeline that does the pulling, it's not guaranteed to execute on all daemons in a distributed setup.

Reply:
I'm not sure if this current PR would hit dockerhub rate limits as-is, but in general I think image pulling should be cleaned up.

I'm just not sure if it should be done before or after this PR.

Reply:
I don't think exact tags is really maintainable without automation that also continuously updates the versions as some images publish very frequently.

Reply:
We use a combination of a static runner, and autoscaler that will create/destroy runners as needed.

Reply:
> We use a combination of a static runner, and autoscaler that will create/destroy runners as needed.

Ok, sounds like we are not really in control of the docker image storage in such a setup, right?